src/django/pfb_network_connectivity/settings.py

```diff
@@ -222,7 +222,7 @@
         'rest_framework.authentication.TokenAuthentication',
         'rest_framework.authentication.SessionAuthentication',
     ),
-    'DEFAULT_PAGINATION_CLASS': 'rest_framework.pagination.LimitOffsetPagination',
+    'DEFAULT_PAGINATION_CLASS': 'pfb_network_connectivity.pagination.OptionalLimitOffsetPagination',
```

Comment:
We might not want to make this the default. Or, be careful to override with the required-limit version for endpoints that might be overloaded trying to fetch all results (thinking of the analysis jobs endpoint in particular).